### PR TITLE
Expose I2C addr

### PR DESCRIPTION
**Pull request by TheButlah**
I wish to write some bytes manually to the device, but I don't know its address.

### CODE REVIEW

**Review by eldruin — COMMENTED** [review]
This change is fine in itself. Do you actually need the function to be `const`?
Also, what are you interested in supporting?
It would probably be better to add it directly to the driver.

**Comment by TheButlah** [comment]
I don't explicity need it to be const right now, but this particular feature has been available in rust for three years now, I believe. Surely thats sufficient for the minimum rust version to be bumped to that (I see that CI is pinned to a much older version)

> It would probably be better to add it directly to the driver.

Isn't this the driver? Apologies, im a bit new to embedded, so my terminology could just be wrong.

> Also, what are you interested in supporting?

like broader picture or specifically in this PR? Broader picture I want to do sensor fusion on the bmi160. This requires having control over FSR, ODR, etc, which are PRs I can submit as I write and test them.

**Comment by eldruin** [comment]
> I don't explicity need it to be const right now, but this particular feature has been available in rust for three years now, I believe. Surely thats sufficient for the minimum rust version to be bumped to that (I see that CI is pinned to a much older version)

Bumping the MSRV would be fine but only to make this one trivial function `const` I consider it not worth it.
I can surely bump it to whatever necessary, though. Could you remove the qualifier?

> > It would probably be better to add it directly to the driver.
> 
> Isn't this the driver? Apologies, im a bit new to embedded, so my terminology could just be wrong.

Yes, that phrase referred to the previous one. I meant whatever you want to do by talking directly to the device via its address would be better added to the driver. Sorry for the confusion.

> like broader picture or specifically in this PR? Broader picture I want to do sensor fusion on the bmi160. This requires having control over FSR, ODR, etc, which are PRs I can submit as I write and test them.

That sounds good. You can have a look at my other drivers if you need inspiration: e.g. [accelerometer drivers](https://github.com/eldruin?tab=repositories&q=accelerometer&type=&language=&sort=).
The [KXCJ9](https://github.com/eldruin/kxcj9-rs) is a rather simple accelerometer driver. The [LSM303AGR](https://github.com/eldruin/lsm303agr-rs) is a more complex example of accelerometer/magnetometer combination.

File: src/interface.rs [hunk 1]
```diff
@@ -39,7 +39,8 @@ impl Default for SlaveAddr {
 }
 
 impl SlaveAddr {
-    pub(crate) fn addr(self) -> u8 {
+    /// The numerical address of the `SlaveAddr`
+    pub const fn addr(self) -> u8 {
```
Comment (TheButlah) [inline src/interface.rs:43]:
```suggestion
    pub  fn addr(self) -> u8 {
```

File: src/interface.rs [hunk 2]
```diff
@@ -39,7 +39,8 @@ impl Default for SlaveAddr {
 }
 
 impl SlaveAddr {
-    pub(crate) fn addr(self) -> u8 {
+    /// The numerical address of the `SlaveAddr`
```
Comment (eldruin) [inline src/interface.rs:42]:
```suggestion
    /// The numerical address of the device
```

**Review by eldruin — COMMENTED** [review]
Looks good to me, thanks!

**Comment by eldruin** [comment]
I have released this in version 0.1.1

**Pull request merged**